### DISCUSSION TITLE
Quotes in language files

### DISCUSSION THREAD

**Discussion by counterpoint** [joomla/joomla-cms#33110] [Q&A] [answered]
I am seeing conflicting things. There are discussions that imply language files are processed using the PHP function parse_ini_file. There is also documentation that includes a demonstration of escaping quotes at https://docs.joomla.org/Creating_a_language_definition_file. 

However, as I understand it, the PHP function will not handle escaped quotes. Am I mistaken? Is there a solution to putting quotes in language files? If there is a restriction of no quotes in language files, that is a massive loss of functionality.

**Comment by infograf768** [comment, marked as answer]
```
; \ is used to escape a value.
newline_is = "\\n" ; results in the string "\n", not a newline character.
with quotes = "She said \"Exactly my point\"." ; Results in a string with quote marks in it.
```
in https://www.php.net/manual/en/function.parse-ini-file.php

**Reply by counterpoint** [reply]
Thanks very much. I confused myself. Somewhere I read that the symbols (the left hand side) should be enclosed in quotes, which creates a problem. There are also some old reports of PHP parse_ini_file failing to support escaped quotes. But you are quite correct.

**Reply by richard67** [reply]
@counterpoint So the above comment by infograf768 answers your question?

**Reply by counterpoint** [reply]
Yes thanks, I have marked it so.
